### PR TITLE
Untie TextBox from Vec2, Move TextBox and TimerChain to ClientData

### PR DESCRIPTION
**Pull request by dimitry-**
Use std::pair instead, this lets us move TextBox/TimerChain to ClientData
component. They no longer depend on specific UI implementation.

We want to unitie data used by UI from QT/GL and any other specific implementations of UI
so that we can save it to disk in the future.

Test: builds

### CODE REVIEW

File: src/OrbitGl/TrackManagerTest.cpp [hunk 1]
```diff
@@ -16,27 +16,6 @@ using orbit_client_protos::TimerInfo;
 
 namespace orbit_gl {
 
-class UnitTestTrack : public Track {
```
Comment (karupayun) [inline src/OrbitGl/TrackManagerTest.cpp:19]: @reichlfl I thing you add this UnitTestTrack that it's not being used I suppose. Do you have any complain with this?

**Review by karupayun — APPROVED** [review]
LGTM Dimitry, but I think that you should add some context about why this move to ClientData. 

Just so you know, I have the plan of removing TextBox and constructing them directly when drawing based on timer_info. Let's sync up later today.

File: src/OrbitGl/TrackManagerTest.cpp (hunk 1 above)
Comment (reichlfl) [reply src/OrbitGl/TrackManagerTest.cpp:19]: Ah, right, that may no longer be needed. I thought I'd need it for the tests but ended up using the regular tracks, so removing this is fine.

File: src/ClientData/include/ClientData/TextBox.h [hunk 2]
```diff
@@ -27,31 +24,32 @@ class TextBox {
...
-  const Vec2& GetSize() const { return size_; }
-  const Vec2& GetPos() const { return pos_; }
+  [[nodiscard]] const std::pair<float, float>& GetSize() const { return size_; }
+  [[nodiscard]] const std::pair<float, float>& GetPos() const { return pos_; }
 
-  const std::string& GetText() const { return text_; }
+  [[nodiscard]] const std::string& GetText() const { return text_; }
   void SetText(std::string text) { text_ = std::move(text); }
 
   const orbit_client_protos::TimerInfo& GetTimerInfo() const { return timer_info_; }
```
Comment (florian-kuebler) [inline src/ClientData/include/ClientData/TextBox.h:36]: [[nodiscard]], I you are already updating those.

File: src/OrbitGl/CMakeLists.txt [hunk 3]
```diff
@@ -60,15 +60,14 @@ target_sources(
          ScopeTree.h
          ShortenStringWithEllipsis.h
          StatusListener.h
-         TextBox.h
+        ../ClientData/include/ClientData/TextBox.h
```
Comment (florian-kuebler) [inline src/OrbitGl/CMakeLists.txt:63]: Shouldn't you include the ClientData project instead?

File: src/OrbitGl/ThreadTrack.cpp [hunk 4]
```diff
@@ -368,12 +371,12 @@ void ThreadTrack::SetTimesliceText(const TimerInfo& timer_info, float min_x, flo
   }
 
   const Color kTextWhite(255, 255, 255, 255);
-  const Vec2& box_pos = text_box->GetPos();
-  const Vec2& box_size = text_box->GetSize();
-  float pos_x = std::max(box_pos[0], min_x);
-  float max_size = box_pos[0] + box_size[0] - pos_x;
+  const auto& box_pos = text_box->GetPos();
```
Comment (florian-kuebler) [inline src/OrbitGl/ThreadTrack.cpp:374]: As above. Decide yourself, whether to use auto here.

File: src/OrbitGl/ThreadTrack.cpp [hunk 5]
```diff
@@ -520,28 +524,30 @@ void ThreadTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t
...
       if (text_box.End() <= next_pixel_start_time_ns) continue;
       ++visible_timer_count_;
 
       Color color = GetTimerColor(text_box, draw_data);
       std::unique_ptr<PickingUserData> user_data = CreatePickingUserData(*batcher, text_box);
 
       ResizeTextBox(draw_data, time_graph_, world_timer_y, box_height_, &text_box);
-      const Vec2& pos = text_box.GetPos();
-      const Vec2& size = text_box.GetSize();
+      const auto& pos = text_box.GetPos();
```
Comment (florian-kuebler) [inline src/OrbitGl/ThreadTrack.cpp:535]: And here.

**Review by florian-kuebler — APPROVED** [review]
Despite minor things, LGTM. Thanks!

File: src/OrbitGl/CMakeLists.txt (hunk 3 above)
Comment (dimitry-) [reply src/OrbitGl/CMakeLists.txt:63]: right - I was looking for this one and missed somehow.

File: src/OrbitGl/GpuDebugMarkerTrack.cpp [hunk 6]
```diff
@@ -81,18 +82,18 @@ void GpuDebugMarkerTrack::SetTimesliceText(const TimerInfo& timer_info, float mi
   }
 
   const Color kTextWhite(255, 255, 255, 255);
-  const Vec2& box_pos = text_box->GetPos();
-  const Vec2& box_size = text_box->GetSize();
-  float pos_x = std::max(box_pos[0], min_x);
-  float max_size = box_pos[0] + box_size[0] - pos_x;
+  const auto& box_pos = text_box->GetPos();
```
[the remark this answers is not in the document]
Comment (dimitry-) [reply src/OrbitGl/GpuDebugMarkerTrack.cpp:85]: we might want to change pair of floats to something else in near future - I prefer to keep auto here.

File: src/ClientData/include/ClientData/TextBox.h (hunk 2 above)
Comment (dimitry-) [reply src/ClientData/include/ClientData/TextBox.h:36]: done.

**Pull request merged**